tcms/testplans/static/testplans/js/get.js

```diff
@@ -19,36 +24,45 @@ $(document).ready(function() {
                 testCases.not_confirmed.push(data[i]);
             }
         }
-        drawTestCases(testCases.confirmed);
+        drawTestCases(testCases.confirmed, testPlanId, permissions);
         treeViewBind();
     });
 
 });
 
-function drawTestCases(testCases) {
+function drawTestCases(testCases, testPlanId, permissions) {
     var container = $('#confirmed-testcases'),
         noCasesTemplate = $('#no_test_cases'),
         testCaseRowDocumentFragment = $('#test_case_row')[0].content;
 
     if (testCases.length > 0) {
         for (var i = 0; i < testCases.length; i++) {
-            container.append(getTestCaseRowContent(testCaseRowDocumentFragment.cloneNode(true), testCases[i]));
+            container.append(getTestCaseRowContent(testCaseRowDocumentFragment.cloneNode(true), testCases[i], permissions));
+        }
+        attachEvents(testPlanId, permissions);
+        if (permissions['perm-change-testcase']) {
```

Comment:
Same comment about permission check as below.  If the user doesn't have permissions then the underlying menu items will not be available.